### DISCUSSION TITLE
Clearing custom errors

### DISCUSSION THREAD

**Discussion by mkallies** [react-hook-form/react-hook-form#8257] [Q&A] [answered]
If I have a custom error, such as one returned from an API, how do I go about clearing it when form input changes?

For example I have a login form with email and password. User submits and server returns an error. I set a custom error on the form. I now change either my email or password input and click submit, however I am blocked. What's the best way to clear custom errors whenever there is a change within the form?

**Comment by ritikbanger** [comment, marked as answer]
` clearError()` need to invoked manually to remove that [custom error](https://react-hook-form.com/api/useform/clearerrors). Here is a [codesandbox](https://codesandbox.io/s/react-hook-form-v7-ts-clearerrors-w3ymx) for example.

**Comment by mkallies** [comment]
Thanks for writing up an example! Unfortunately having a button to clear a custom error breaks UX. I've added an `onChange` function to the `register` options and if there's any server errors I'll just clear them

**Comment by Moshyfawn** [comment]
If you enter a different value, the manual error disappears: see [this example](https://codesandbox.io/s/react-hook-form-v7-ts-clearerrors-forked-xllgep?file=%2Fsrc%2Findex.tsx)

**Reply by mkallies** [reply]
Hey thanks for the reply @Moshyfawn. My case was with a custom error. I wish there was an option to clear a custom error when there's an input change. Had to do something like this:
```
          {...register('someInput', {
            onChange: () => {
              if (hasServerErrors) {
                clearErrors('serverError')
              }
            },
            required: 'Please enter something',
          })}
```

**Reply by Moshyfawn** [reply]
What do you mean by "custom error"? An error set with the `setError` form method?

[In my example](https://codesandbox.io/s/react-hook-form-v7-ts-clearerrors-forked-xllgep?file=%2Fsrc%2Findex.tsx), you can submit a valid form _(all three fields are non-empty)_. The submission will result in a custom error set to the `firstName` field. Start typing in that field; the error is cleared

**Reply by mkallies** [reply]
In your example I'd have an error returned by an API with `setError('apiError')` is not related to any form field

**Reply by Moshyfawn** [reply]
Ah, your use-case is described in #7661 feature request.

RHF doesn't support "global" errors ATM
